packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -43,16 +46,41 @@ library LibProposing {
             msg.sender != resolver.resolve("solo_proposer", false)
         ) revert L1_NOT_SOLO_PROPOSER();
 
-        if (txList.length > config.maxBytesPerTxList) revert L1_TX_LIST();
+        if (input.txListEnd <= input.txListStart) revert L1_TX_LIST_RANGE();
 
         if (
             input.beneficiary == address(0) ||
             input.gasLimit > config.blockMaxGasLimit
         ) revert L1_INVALID_METADATA();
 
-        // We need txListHash as with EIP-4844, txList is no longer
-        // accssible to EVM.
-        if (input.txListHash != keccak256(txList)) revert L1_TX_LIST_HASH();
+        uint64 _now = uint64(block.timestamp);
+        uint32 _size = uint32(txList.length);
+
+        if (_size == 0) {
+            // This blob shall have been submitted earlier
+            TaikoData.TxListInfo memory info = state.txListInfo[
+                input.txListHash
+            ];
+
+            if (
+                info.size == 0 ||
+                info.validSince + config.txListCacheExpiry < _now
+            ) revert L1_TX_LIST_NOT_EXIST();
+
+            if (input.txListEnd > info.size) revert L1_TX_LIST_RANGE();
+        } else {
+            if (_size > config.maxBytesPerTxList) revert L1_TX_LIST();
+            if (input.txListEnd > _size) revert L1_TX_LIST_RANGE();
```

Comment:
@Brechtpd as of now, the `txListStart` and `txListEnd` are byte-wise indices. Will use them as the transaction indices easier for circuits to implement?

Reply:
Byte indices will be easier, otherwise we have to be able to decode all data, which may not even be possible because it may contain other data as well which should not impact the given range.